parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -459,26 +454,44 @@ public Builder withMaxBloomFilterBytes(int maxBloomFilterBytes) {
     }
 
     /**
-     * Set Bloom filter column names and expected NDVs.
+     * Set Bloom filter NDV (number of distinct values) for the specified column.
+     *
+     * @param columnPath the path of the column (dot-string)
+     * @param ndv the NDV of the column
+     *
+     * @return this builder for method chaining
+     */
+    public Builder withBloomFilterNDV(String columnPath, long ndv) {
+      Preconditions.checkArgument(ndv > 0, "Invalid NDV for column \"%s\": %d", columnPath, ndv);
+      this.bloomFilterNDVs.withValue(columnPath, ndv);
+      // Setting and NDV for a column implies writing a bloom filter
+      this.bloomFilterEnabled.withValue(columnPath, true);
+      return this;
+    }
+
+    /**
+     * Enable or disable the bloom filter for the columns not specified by
+     * {@link #withBloomFilterEnabled(String, boolean)}.
      *
-     * @param columnToNDVMap the columns which has bloom filter enabled.
+     * @param enabled whether bloom filter shall be enabled
      *
      * @return this builder for method chaining
      */
-    public Builder withBloomFilterColumnToNDVMap(Map<String, Long> columnToNDVMap) {
-      this.bloomFilterColumnExpectedNDVs = columnToNDVMap;
+    public Builder withBloomFilterEnabled(boolean enabled) {
```

Comment:
Can we document the settings of their relations? Here are my understandings. 
1. 'withBloomFilterEnabled(boolean enabled)' would set it globally, which can be overridden by column specific setting of 'withBloomFilterEnabled(String columnPath, boolean enabled)' or 'withBloomFilterNDV(String columnPath, long ndv)'. 2. Now 'withBloomFilterNDV(String columnPath, long ndv)' will override 'withBloomFilterEnabled(String columnPath, boolean enabled)' implicitly or vice versa explicitly. 

I feel it is a bit complicated to use given the above relations. It is also my understanding that NDV should only present when bloomFitler is enabled. If that is the case, can we just use NDV null or not to indicate enabling/disabling to simplify the settings.

Reply:
Bloom filters are not written by default for any columns. You may turn it on for the different columns. The "global" setting is just a shortcut to turn bloom filters on for all columns (and you may turn off for some of them).
NDV can be set for a specific column. If not set (but the bloom filter is turned on for that column) bloom filter can still be written using a default calculation. It is another short cut that if the NDV is set for a column then the user obviously want to write a bloom filter for that column so we set the related flag.

Reply:
We were on the same page except for the 'default calculation' for NDV. Now I understand better. But my concern is still that it is a little complicated to use these 3 settings including the 2 shortcuts.  How about we rename the two column-specific setting methods like withEnableBloomFilterWithNDV() and withEnableBloomFilterWithDefaultNDV()?

Reply:
There is no default NDV. We calculate the optimal size of the bloom filter based on the NDV if it is set and use the maximum size if it is not. (See the related code [here](https://github.com/apache/parquet-mr/pull/763/files#diff-c50835e13bd6059132963b559e57303bR91-R97))

I've tried to explain the current behavior of the properties more precisely in the method comments.

Reply:
Sounds good